client/src/TemplateQuestionsPage.js

```diff
@@ -22,11 +23,19 @@ const scriptNameQuestion = {
 
 function TemplateQuestionsPage() {
   const { templateId } = useParams();
+  const ref = useRef();
+
+  // During page transitions, useParams may return undefined
+  // In these cases, the previously stored reference value is used
+  if (templateId) {
+    ref.current = templateId
+  }
```

Comment:
This would more be idiomatically and concisely written as

```
const ref = useRef(templateId)
```

`useRef` will set the first value it is given to `ref.current` and ignore the value given in all subsequent calls.